eng/Generation.psm1

```diff
@@ -50,6 +50,59 @@ function AutoRest-Reset()
     Invoke "$script:autoRestBinary --reset"
 }
 
+function Invoke-Cadl($baseOutput, $projectName, $sharedSource="", $fast="", $debug="")
+{
+    $baseOutput = Resolve-Path -Path $baseOutput
+    $baseOutput = $baseOutput -replace "\\", "/"
+    $outputPath = Join-Path $baseOutput "Generated"
+    $outputPath = $outputPath -replace "\\", "/"
+    #clean up
+    if (Test-Path $outputPath) {
+        Remove-Item $outputPath/* -Include *.* -Exclude *Configuration.json*
+    }
+    # emit cadl json
+    $repoRootPath = Join-Path $PSScriptRoot ".."
+    $repoRootPath = Resolve-Path -Path $repoRootPath
+    Push-Location $repoRootPath
+    # node node_modules\@cadl-lang\compiler\dist\core\cli.js compile --output-path $outputPath "$baseOutput\$projectName.cadl" --emit @azure-tools/cadl-csharp
+    $emitCommand = "node node_modules/@cadl-lang/compiler/dist/core/cli.js compile --output-path $outputPath $baseOutput/$projectName.cadl --emit @azure-tools/cadl-csharp"
+    Invoke $emitCommand
+    Pop-Location
+    
+    $dotnetArguments = $debug ? "--no-build --debug" : "--no-build" 
+    $command = "dotnet run --project $script:AutoRestPluginProject $dotnetArguments --standalone $outputPath"
+    Invoke $command
+    Invoke "dotnet build $baseOutput --verbosity quiet /nologo"
+}
+
+function Invoke-CadlSetup()
+{
+    # build emitter
+    $emitterPath = Join-Path $PSScriptRoot ".." "src" "CADL.Extension" "Emitter.Csharp"
+    $emitterPath = Resolve-Path -Path $emitterPath
+    Push-Location $emitterPath
+    npm install
+    npm run build
+    npm pack
+    Pop-Location
+
+    # install cadl and emitter
+    $repoRoot = Join-Path $PSScriptRoot ".."
+    $repoRoot = Resolve-Path $repoRoot
+    Push-Location $repoRoot
+    $packages = Get-ChildItem $repoRoot -Filter azure-tools-cadl-csharp-*.tgz -Recurse | Select-Object -ExpandProperty FullName | Resolve-Path -Relative
+    if ($packages) {
+        $package = $packages;
+        if ($packages.count -gt 1) {
+            $package = $packages[0]
+        }
+        npm install $package --save
+    }
+    npm install
+    git checkout $repoRoot/package.json
+    git checkout $repoRoot/package-lock.json
```

Comment:
Is this a temporary solution? And how could we garantee there must be git installed.

Reply:
We need to install an local emitter version, and it will update the package.json, package-lock.json, and as a result, it cause the code-out-of-update failure. There is no way to avoid updating them.